test/yugabyte_test.go

```diff
@@ -0,0 +1,282 @@
+		outputStatus, errStatus := ssh.CheckSshCommandE(t, sshSession, statusCodeCheckCmd)
+		if errStatus != nil {
+			return "", errStatus
+		}
+
+		outputBody, errBody := ssh.CheckSshCommandE(t, sshSession, bodyURLCmd)
+		if errBody != nil {
+			return "", errBody
+		}
+
+		if !strings.Contains(string(outputBody), "YugabyteDB Version") {
+			return "", fmt.Errorf("Failed to find the 'YugabyteDB Version' in output\nStatus Code: %s, Out from Master UI: %s", outputStatus, outputBody)
+		}
+
+		logger.Logf(t, "Master UI Status Code:- %s", outputStatus)
+		return "", nil
+	})
+}
+
+// Verify the status of Tserver UI
+func testYugaByteTserverURL(t *testing.T, terraformOptions *terraform.Options, tserverUI string, sshSession ssh.Host) {
+	statusCodeCheckCmd := fmt.Sprintf("curl -o /dev/null -s -w \"%%{http_code}\\n\" http://%s:9000", tserverUI)
+	bodyURLCmd := fmt.Sprintf("curl http://%s:9000", tserverUI)
+	retry.DoWithRetry(t, "Checking YB Tserver UI", maxRetries, timeBetweenRetries, func() (string, error) {
+		outputStatus, errStatus := ssh.CheckSshCommandE(t, sshSession, statusCodeCheckCmd)
+		if errStatus != nil {
+			return "", errStatus
+		}
+
+		outputBody, errBody := ssh.CheckSshCommandE(t, sshSession, bodyURLCmd)
+		if errBody != nil {
+			return "", errBody
+		}
+
+		if !strings.Contains(string(outputBody), "YugabyteDB") {
+			return "", fmt.Errorf("Failed to find the 'YugabyteDB' in output\nStatus Code: %s, Out from Tserver UI: %s", outputStatus, outputBody)
+		}
+
+		logger.Logf(t, "Tserver UI Status Code:- %s", outputStatus)
+		return "", nil
+	})
+}
+
+func testYugaByteSSH(t *testing.T, terraformOptions *terraform.Options, sshSession ssh.Host) {
+	sshCommand := fmt.Sprintf("echo \"Hello Terratest! I am $(whoami) user at $(hostname)\"")
+	retry.DoWithRetry(t, "Attempting to SSH", maxRetries, timeBetweenRetries, func() (string, error) {
+		output, err := ssh.CheckSshCommandE(t, sshSession, sshCommand)
+		if err != nil {
+			return "", err
+		}
+		logger.Logf(t, "SSH Command Output: %s\n", output)
+		return "", nil
+	})
+}
+
+// Verify the YSQL API
+func testYugaByteYSQLSH(t *testing.T, terraformOptions *terraform.Options, privateIP string, sshUser string, sshSession ssh.Host) {
+	commandConnectYSQLSH := fmt.Sprintf("cd /home/%s/yugabyte-db/tserver && ./bin/ysqlsh --echo-queries -h %s -c \\\\conninfo", sshUser, privateIP)
+	retry.DoWithRetry(t, "Conecting YSQLSH", maxRetries, timeBetweenRetries, func() (string, error) {
+		output, err := ssh.CheckSshCommandE(t, sshSession, commandConnectYSQLSH)
+		if err != nil {
+			return "", err
+		}
+		logger.Logf(t, "Output of Ysql command :- %s", output)
+		return "", nil
+	})
+}
+
+// Verify the YCQL API
+func testYugaByteCQLSH(t *testing.T, terraformOptions *terraform.Options, privateIP string, sshUser string, sshSession ssh.Host) {
+	commandConnectCQLSH := fmt.Sprintf("cd /home/%s/yugabyte-db/tserver && ./bin/ycqlsh %s 9042 --execute 'SHOW HOST'", sshUser, privateIP)
```

Comment:
minor: doesn't this command work when run with the full path instead of a cd && ycqlsh?